tools/deserialize-proto/main.go

```diff
@@ -0,0 +1,73 @@
+package main
+
+import (
+	"bufio"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"log"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/golang/protobuf/proto"
+	flag "github.com/spf13/pflag"
+)
+
+var protobufType *string = flag.String("type", "", "name of protobuf, e.g., storage.Alert")
+
+func unmarshalProto[T proto.Unmarshaler](t T, b []byte) error {
+	return t.Unmarshal(b)
+}
+
+func main() {
+	flag.Parse()
+
+	if *protobufType == "" {
+		log.Fatal("must provide --type")
+	}
+
+	messageType, ok := typeRegistry[*protobufType]
+	if !ok {
+		log.Fatalf("%s is an invalid storage type", *protobufType)
+	}
+
+	var text string
+
+	reader := bufio.NewScanner(os.Stdin)
+	for reader.Scan() {
+		text = reader.Text()
+		if len(text) == 0 {
+			break
+		}
+
+		// It's not clear why we need to both unquote *and* prepend 0A but it works ¯\_(ツ)_/¯
+		s, err := strconv.Unquote(fmt.Sprintf("\"%s\"", text))
+		if err != nil {
+			log.Fatalf("error while unquote the serialized text, text = %s, err = %v", text, err)
+		}
+
+		s = "0A" + strings.TrimSpace(s)
```

Comment:
The `0A` prepend is interesting, I was removing `\x` from the beginning of the data before running `hex.DecodeString()`, like this:

```
	for dataB[0] == '\\' || dataB[0] == 'x' {
		dataB = dataB[1:]
	}

	protoBytes, err := hex.DecodeString(string(dataB))
	if err != nil {
		t.Error(err)
	}
```
But if it works, it works!

Reply:
I did the whole `strconv.Unquote` business in case there were any strings or characters we're not accounting for, but maybe that's an (evil) premature optimization. I'm down for either implementation. Probably gonna merge this as-is, but please feel free to change it 🙂